CONVENTIONS.md

```diff
@@ -89,15 +89,57 @@ metal3-io](https://github.com/metal3-io/metal3-docs/blob/master/design/bare-meta
 
 OpenShift APIs follow the [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md).
 
-### Operators
+### Cluster Conventions 
+
+A number of rules exist for individual components of the cluster in order to ensure
+the overall goals of the cluster can be achieved. These ensure OpenShift is resilient,
+reliable, and consistent for administrators.
+
+#### Use Operators To Manage Components
 
 The OpenShift project is an early adopter of, and makes extensive use of, [the
-operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/),
-and so it is incumbent on us to establish some conventions around operators.
+operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) and
+it is used to manage all elements of the cluster.
+
+* All components are managed by operators
+  * All cluster component operators are expected to deploy, reconfigure, self-heal, and report status about its health back to the cluster
+  * The cluster-version-operator (CVO) provides a minimum lifecycle for the core operators that
+    are required to get OLM running, and OLM runs all other operators
+  * The core operators are updated together and represent the minimum set of function for an OpenShift cluster
+  * Any operator that is an optional part of OpenShift or has a different lifecycle than the core OpenShift release must run under OLM
+* All configuration is exposed as API resources on cluster
+  * Immutable configuration is minimal and exposed as "status" fields of a cluster, set during installation
+  * All other configuration is expected to work like normal Kube objects - changeable at any time, validated, and any failures reported via the status fields of that API
+* Every component must remain available to consumers without disruption during upgrades and reconfiguration
+  * See the next section for more details
+
+
+#### Upgrade and Reconfiguration
+
+* Every component must remain available to consumers without disruption during upgrades and reconfiguration
+  * Disruption caused by node restarts is allowed but must respect disruption policies (PodDisruptionBudget) and should optimized to reduce total disruption
+    * Administrators may minimize 
```

Comment:
was this the end of prior sentence or start of new?

Reply:
finished incomplete thought